common/types.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * A mapping between libraries and the addresses to which they were deployed.
+ *
+ * Containing support for two level configuration.
+ */
+export interface LibraryAddresses {
+  [libraryName: string]: string | { [location: string]: string };
+}
+
+/**
+ * A mapping between libraries and lists of placeholder instances present in their hex-encoded bytecode.
+ * For each placeholder its length and the position of the first character is stored.
+ */
+export interface LinkReferences {
+  [libraryName: string]: Array<{ start: number, length: number }>;
```

Comment:
After taking a closer look at `findLinkReferences` I see that this is not necessarily the library name. In some specific cases it could be but it can also be truncated and in case of new-style references it will be the hash with `$` markers (i.e. `$cb901161e812ceb78cfe30ca65050c4337$`)

It's what I referred to in other places as *library label*. I.e. just the placeholder without the `__` markers and not padded with extra `_` characters for full 40 chars.
```suggestion
  [libraryLabel: string]: Array<{ start: number, length: number }>;
```

Unfortunately it's hard to keep the terminology here straight because the current code uses placeholder, name and label interchangeably sometimes.